[Failing Test] pkg/volume/volume_linux_test.go

### Which jobs are failing?

NONE

### Which tests are failing?

- TestSetVolumeOwnershipOwner/*fsGroup=3000
- TestSetVolumeOwnershipOwner/symlink

### Since when has it been failing?

Since PR #130398 

### Testgrid link

_No response_

### Reason for failure (if possible)

-  volume_linux_test.go:592: for *fsGroup=3000 error verifying permissions with: invalid owner on /tmp/volume_linux_ownership3055383160/file.txt
- volume_linux_test.go:592: for symlink error verifying permissions with: invalid owner on /tmp/volume_linux_ownership3642780172/file_link.txt

### Anything else we need to know?

Two test cases started to fail after PR #130398, not sure if this [line](https://github.com/kubernetes/kubernetes/pull/130398#discussion_r1982400934) is the one causes the issue or not. We may need to revert or fix the test to work regardless of which user runs the test.

Error:
```
{Failed  === RUN   TestSetVolumeOwnershipOwner/*fsGroup=3000
    volume_linux_test.go:592: for *fsGroup=3000 error verifying permissions with: invalid owner on /tmp/volume_linux_ownership3055383160/file.txt
--- FAIL: TestSetVolumeOwnershipOwner/*fsGroup=3000 (0.60s)

=== RUN   TestSetVolumeOwnershipOwner/symlink
    volume_linux_test.go:592: for symlink error verifying permissions with: invalid owner on /tmp/volume_linux_ownership3642780172/file_link.txt
--- FAIL: TestSetVolumeOwnershipOwner/symlink (0.90s)

=== RUN   TestSetVolumeOwnershipOwner
--- FAIL: TestSetVolumeOwnershipOwner (1.51s)
}
```

### Relevant SIG(s)

/sig

--- a/pkg/volume/volume_linux_test.go
+++ b/pkg/volume/volume_linux_test.go
@@ -332,16 +332,32 @@ func TestSetVolumeOwnershipMode(t *testing.T) {
 func TestProgressTracking(t *testing.T) {
 	alwaysApplyPolicy := v1.FSGroupChangeAlways
 	var expectedGID int64 = 9999
+
+	//  capture original variable
+	originalfilePermissionChangeFunc := filePermissionChangeFunc
+	originalProgressReportDuration := progressReportDuration
+	originalfirstEventReportDuration := firstEventReportDuration
+
 	var permissionSleepDuration = 5 * time.Millisecond
-	// how often to report the events
+
+	// Override how often progress is reported
 	progressReportDuration = 200 * time.Millisecond
+	// Override when first event about progress is reported
 	firstEventReportDuration = 50 * time.Millisecond
 
-	filePermissionChangeFunc = func(filename string, fsGroup *int64, _ bool, _ os.FileInfo) error {
-		t.Logf("Calling file permission change for %s with gid %d", filename, *fsGroup)
+	// Override how permission change is applied, so as to artificially slow
+	// permission change
+	filePermissionChangeFunc = func(filename string, fsGroup *int64, readonly bool, info os.FileInfo) error {
 		time.Sleep(permissionSleepDuration)
+		originalfilePermissionChangeFunc(filename, fsGroup, readonly, info)
 		return nil
 	}
+	t.Cleanup(func() {
+		filePermissionChangeFunc = originalfilePermissionChangeFunc
+		progressReportDuration = originalProgressReportDuration
+		firstEventReportDuration = originalfirstEventReportDuration
+	})
+
 	pod := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "pod1",

